Create .NET docs What's new section

BillWagner opened the pull request:
Contributes to #16189 

First pass automation for "What's new" page.

It turned out that automating it (in part) took less time than putting it together by hand.

I'd like everyone I tagged to review this. I already recognized a few false positives in the "what's new" article, and at least one MS employee in the contributor list.

False positives: Some articles had significant formatting changes, but aren't really new. Some others had large amount of changes, but really aren't noteworthy.

I'll merge this after approval, and merging #16248. I'll add a "highlight" for the What's new page.

- [What's new landing page internal review link](https://review.docs.microsoft.com/en-us/dotnet/whats-new/index?branch=pr-en-us-16197).
- [What's new in November2019](https://review.docs.microsoft.com/en-us/dotnet/whats-new/2019-11?branch=pr-en-us-16197)

BillWagner committed:
first draft

BillWagner committed:
interim checkin

BillWagner committed:
First take at the automated generation.

BillWagner committed:
fix markdown lint warning.

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+---
+title: ".NET Docs What's New (November 2019)"
```
```suggestion
title: ".NET docs what's new (November 2019)"
```

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+ms.date: 2019/12/13
+---
+
+# .NET Docs What's New (November 2019)
```
```suggestion
# .NET docs what's new (November 2019)
```

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+
+# .NET Docs What's New (November 2019)
+
+Welcome to what's new in .NET docs for November 2019. Listed below are some of the recent docs we've published.
```
The title implies that this is for docs published in November, but the list seems too long for that. Also this ("November") seems to conflict with the contributor list which says it's "since our last release notes".  Suggestion is to clarify the period covered, and make the period consistent for docs and contributor list.

Also, it would also be ideal to call out differently docs that are new as opposed to just revised. To be even more useful, revised docs would have a short blurb summarizing what was done to them.

BillWagner replied:
@MaximRouiller How did you tell the difference between new and updated docs? I could not find this info in the API values.

BillWagner replied:
I had to use the REST API, instead of the GraphQL API.

MaximRouiller replied:
New pages were using ⭐️ emojis as a way to differentiate from purely modified pages. 

In my code, new pages were generated by a single command line, the modified were from a different command line. I haven't used the API so I would need to investigate that issue.

BillWagner replied:
I love the :star: emoji. I didn't use it to account for accessibility concerns. In a future PR, I would add it, if we can solve that with alt-text.

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+
+## C# language
+
+- [https://raw.githubusercontent.com/dotnet/docs/master/docs/csharp/index.yml](../docs/csharp/index.yml)
```
This doesn't seem like it belongs in this list.

BillWagner replied:
same bug.

BillWagner committed:
fix build warnings.

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+- [dotnet-trace tool - .NET Core](../docs/core/diagnostics/dotnet-trace.md)
+- [.NET Core SDK and runtime dependencies - .NET Core](../docs/core/install/dependencies.md)
+- [Check installed .NET Core versions on Windows, Linux, and macOS - .NET Core](../docs/core/install/how-to-detect-installed-versions.md)
+- [https://raw.githubusercontent.com/dotnet/docs/master/docs/core/install/includes/package-manager-heading-hack-pkgname.md](../docs/core/install/includes/package-manager-heading-hack-pkgname.md)
```
Why does this have a URL in the link text?

BillWagner replied:
That was a bug. It's fixed now.

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+
+Welcome to what's new in .NET docs for November 2019. Listed below are some of the recent docs we've published.
+
+## .NET Core
```
I see .NET Core and .NET Framework, but not .NET, i.e., .NET Standard. Was the .NET guide overlooked, or were there no significant changes in it?

BillWagner replied:
This time, I didn't find any significant changes.

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+## F# language
+
+- [Get Started with F# in Visual Studio Code](../docs/fsharp/get-started/get-started-vscode.md)
+- [https://raw.githubusercontent.com/dotnet/docs/master/docs/fsharp/index.yml](../docs/fsharp/index.yml)
```
ditto

BillWagner replied:
same bug. 😞

tdykstra commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,171 @@
+
+We'd like to recognize the following people, who've contributed to.NET docs since our last release notes:
+
+- [Jeej](https://github.com/Jeej). 
```
Would it be feasible to order by number of contributions, most to least? If not, I suggest we put those without names at the end of the list instead of the beginning.

BillWagner replied:
I could sort by number of contributions. I'll write that code.

MaximRouiller replied:
Agreed with @tdykstra on the ordering. Maybe add a link to how to contribute to docs and be on that list? 🤔

BillWagner committed:
fix warnings

The fix for <bindings> was done by hand, because I doubt it will be a common occurence.

BillWagner committed:
remove extra blank line.

tdykstra commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,35 @@
+          - text: November 2019
+            url: 2019-11.md
+
+  - title: "Open source contributions"
```
Not clear what this means -- seems like it might mean "community contributions" as opposed to MSFT contributions, but the list doesn't look like that's the intention.

BillWagner replied:
This section is meant to provide a roadmap for new contributors. I'll update the text to make that more clear.

tdykstra commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,35 @@
+      - linkListType: overview
+        links:
+          - text: ".NET Docs repositories overview."
+            url: https://github.com/dotnet/docs/blob/master/README.md
```
This didn't change in November - what's the date range for this what's new list?

BillWagner replied:
For the landing page, I followed the [Xamarin example](https://docs.microsoft.com/xamarin/whats-new/).  They use that section to help people find ways to contribute. I'll change the titles to reflect that better.

BillWagner commented:
Open question:

Because the individual monthly pages are automated, should I add pages for October and September to pick up the updates for the .NET Core 3.0 launch?

tdykstra reviewed (dismissed):
Great start -- I left a few suggestions for you to consider.  I wasn't able to look at staging because it looks like the PR didn't build.

gewarren commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,35 @@
+          - text: November 2019
+            url: 2019-11.md
+
+  - title: "Open source contributions"
```
```suggestion
  - title: Contribute
```

BillWagner replied:
I went with different text, based on Tom's comment. I think it addresses the same concern.

gewarren commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,35 @@
+          - text: ".NET API reference docs Contributor guide"
+            url: https://github.com/dotnet/dotnet-api-docs/blob/master/CONTRIBUTING.md
+          - text: ".NET samples Contributor guide"
+            url: https://github.com/dotnet/samples/blob/master/CONTRIBUTING.md
```
Should we have another card with links to what's new in docs for other docsets, e.g. ASP.NET, Xamarin?

BillWagner replied:
I like that idea. What do you think @davidbritch?

mairaw replied:
Especially because with two cards the layout looks weird.

davidbritch replied:
> I like that idea. What do you think @davidbritch?

Sounds good. Also happy to link to your page from our what's new docs.

BillWagner replied:
Latest commit has this added.

gewarren commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,170 @@
+
+## Contributors
+
+We'd like to recognize the following people, who've contributed to.NET docs since our last release notes:
```
```suggestion
The following people have contributed to .NET docs in the last month. Thank you!
```

gewarren reviewed (commented):
Looking good Bill, thanks for doing this!

BillWagner committed:
Respond to feedback

Also, fix some title casing.

BillWagner commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,35 @@
+  ms.date: 12/12/2019
+
+landingContent:
+  - title: "What's new"
```
Nice!

BillWagner committed:
address feedback

BillWagner committed:
more feedback.

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,170 @@
+- [nschonni](https://github.com/nschonni)  - **Nick Schonning**
+- [SeanKilleen](https://github.com/SeanKilleen)  - **Sean Killeen**
+- [ThePiranha](https://github.com/ThePiranha)  - **Sabah Shariq**
+- [mvelosop](https://github.com/mvelosop)  - **Miguel Veloso**
```
mvelosop, rpetrusha, steveada are/were either internal folks or vendors

MaximRouiller replied:
I think it was a bug as per @BillWagner. As long as they are community members at the time of this update, I'm okay with them appearing on the list.

On top of that, they should not be retroactively removed if they become employees or vendors.

mairaw replied:
The work they’ve done for these updates were not as community contributors so that’s why I was suggesting to remove them.

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,170 @@
+
+## .NET Core
+
+- [Breaking changes, version 2.2 to 3.0 - .NET Core](../core/compatibility/2.2-3.0.md)
```
I'd remove the automated suffix from these or use the H1 like Mark suggested

BillWagner replied:
I did use the H1. I like that a lot better.

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,170 @@
+- [What's new in F# 4.6 - F# Guide](../fsharp/whats-new/fsharp-46.md)
+- [What's new in F# 4.7 - F# Guide](../fsharp/whats-new/fsharp-47.md)
+
+## Contributors
```
Make it clear it's only external so FTEs don't complain they're not being listed

```suggestion
## External contributors
```

BillWagner replied:
I chose "Community contributors" instead of "External". What do you think?

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,170 @@
+
+The following people have contributed to .NET docs in the last month. Thank you!
+
+- [Youssef1313](https://github.com/Youssef1313)  - **Youssef Victor**
```
what's the order here? number of PRs?

BillWagner replied:
Yes, it's by number of PRs.

mairaw commented:
Love it @BillWagner! Left a few comments for you to consider.

BillWagner committed:
add highlight for what's new

Put the what's new content on the Hub page.

BillWagner committed:
Program updates for feedback

1. Use the H1 title instead of the title: metadata.
1. Highligh "new" content.
1. Don't bold contributor names.
1. Update contributor header.
1. Add PR title on updated content.

BillWagner commented:
**Updated status of this PR**

I made numerous programming changes this morning based on the feedback. (See the last commit message). I like the automated parts quite well. Some notes for us as a team:

- I'm using the PR title for the "description of the changes". That's OK, and I think it can be a bit better if we think about the "What's new" page when we write PR titles going forward.
> I tried an experiment using the PR body, but that had too much text in most cases, including the internal review link, related issue, and quite a bit of markdown formatting. It really looked bad.
- I'm using membership it either the "Microsoft" org or the "Microsoft Docs" org to filter MS employees from the contributor list. As Maira points out, that's not perfect: It doesn't filter out Ron, as he's retired. It doesn't filter out vendors, and it misses a few FTEs that haven't joined those orgs. There are three solutions. They all have tradeoffs. What do others think?
  1. I could hard code known GitHub Ids.
  1. We edit these by hand each month.
  1. We define a team that I can filter on in addition to the org membership.
- The biggest question for the team and @MaximRouiller: Is this "close" to the right set of important content? Should I set the changed threshold higher (fewer articles), or lower (more articles)?

tdykstra commented:
Agreed on using PR title and that it'll improve as we give more thought to where it's going to show up when we write the titles.

I vote for hard-coding the GitHub IDs.

I vote for setting the threshold higher.  IF there are significant changes we want to call out that don't make the cut in the automated page creation, we can add them manually.

BillWagner committed:
add a card with other what's new info

The Xamarin docset has a great "What's new" page. That's our most prominent link because it is so well done.

Other areas are the release notes for our key product areas.

BillWagner committed:
threshlod at 50

BillWagner committed:
threshold set at 75

BillWagner committed:
threshold set to 100

BillWagner commented:
> I vote for setting the threshold higher. IF there are significant changes we want to call out that don't make the cut in the automated page creation, we can add them manually.

Check the three latest commits. I set the change threshold at 50, 75, and 100 (from the initial 25). Which do you like best?

MaximRouiller commented:
> * There are three solutions. They all have tradeoffs. What do others think?
>   
>   1. I could hard code known GitHub Ids.
>   2. We edit these by hand each month.
>   3. We define a team that I can filter on in addition to the org membership.

1 would mean a list we have to maintain. 2 would be easy to remove as most employees are in the org and we could train those who aren't to join. 3, I don't quite understand. 

> * The biggest question for the team and @MaximRouiller: Is this "close" to the right set of important content? Should I set the changed threshold higher (fewer articles), or lower (more articles)?

Reviewing this in a few more minutes. Going through the backlog.

BillWagner commented:
@MaximRouiller 

> Reviewing this in a few more minutes. Going through the backlog.

The last four commits show different modification thresholds: 25, 50, 75, 100. That should make it easier to compare.

MaximRouiller commented:
100 is extremely aggressive. 75 is very aggressive, and 50 sounds about right.

New articles being removed, is it a _bug_? If it's new, it should include them whether there is 2 lines or 150 lines. For example, I kind of like 75 but it removes good articles from F# and .NET.

On a side note, we should that number as "movable" throughout this experiment. I say we start at 75)(keeping new articles regardless of size) and see where it leads us.

Too many links may lead to people just ignoring the article. I think 75 sounds about right.

For NuGet, it was set to 15 as they mostly make smaller changes. No idea what to do with this.

MaximRouiller commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,107 @@
+title: ".NET docs what's new (November 2019)"
+description: "What's new in .NET docs in November 2019."
+ms.date: 12/16/2019
+---
```
Will there be a Feedback section at the bottom? If yes, do we need an author for this to tag the right people correctly?

BillWagner replied:
Good catch @MaximRouiller 

@mairaw Could that be the docs team account? Thoughts?

mairaw replied:
Not sure if we can use team aliases for author. Pick someone and add them to the docfx.json file. 😉

MaximRouiller commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,107 @@
+
+## Community contributors
+
+The following people have contributed to .NET docs in the last month. Thank you!
```
Can we add a link on how to contribute to docs? 

https://docs.microsoft.com/contribute/ would sound about right.

BillWagner replied:
I added that link to the landing page, and pointed to the landing page on each sprint what's new page.

BillWagner commented:
> New articles being removed, is it a bug?

Yes, that's a bug. I fixed the code, but I didn't re-run all the commits yet. I'll do that once we settle on the threshold for modified articles.

MaximRouiller commented:
Approved the results. I love what you've done @BillWagner! This is amazing work!

MaximRouiller commented:
Oh and I may owe the .NET Docs team a beverage of their choice. Or maybe I can pay in Maple Syrup? 🤔

mamccrea commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,54 @@
+          - text: ".NET samples contributor guide"
+            url: https://github.com/dotnet/samples/blob/master/CONTRIBUTING.md
+
+  - title: "Related What's new pages"
```
Can we make "What's" lowercase to make the casing more consistent?

mamccrea commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,107 @@
+
+## .NET Core
+
+- ***new!*** [Breaking changes for migration from Version 2.2 to 3.1](../core/compatibility/2.2-3.1.md) - Core breaking changes improvements
```
I'm sorry if this was already answered, but what is considered ***new!***? Does that mean brand new instead of newly updated?

BillWagner replied:
For "new', a PR added the file.

mamccrea reviewed (commented):
This looks great, @BillWagner! I just added two comments/questions. How can we add .NET for Spark to the automation?

BillWagner commented:
@mamccrea 

> How can we add .NET for Spark to the automation?

I *should* be picking those up. Looks like I have a bug to fix. Can you point me to a PR in the November sprint that should be included?

mamccrea commented:
> @mamccrea
> 
> How can we add .NET for Spark to the automation?
> 
> I should be picking those up. Looks like I have a bug to fix. Can you point me to a PR in the November sprint that should be included?

I think this is the only one: https://github.com/dotnet/docs/pull/15840

The others were opened in Nov but merged in Dec.

BillWagner commented:
Thanks @mamccrea That helped me find it.

I'll commit a new update soon that includes a number of areas that got removed by mistake. I'm glad you helped catch that.

BillWagner committed:
fix bug that skipped several sections

*sigh* The bug was that after a section was skipped (like /images), all remaining sections were skipped. That was bad.

BillWagner committed:
respond to feedback.

BillWagner committed:
test run with Sep. and Oct

We're getting closer, so this seemed a good time.

BillWagner committed:
reorder from newest to oldest

Because it's a what's new page.

tdykstra commented:
On the threshold. I agree with @MaximRouiller 

> (keeping new articles regardless of size) 
> Too many links may lead to people just ignoring the article. I think 75 sounds about right.

Additional suggestion: Omit any file that is only updated by a PR that has the default GitHub-generated title of "Update " + filename

BillWagner committed:
Update filters for including changes

- Images are removed automatically
- PRs with a title of "update <file>" are ignored.
- Threshold set at 75.

I think this is ready for final review.

BillWagner committed:
add author metadata.

mairaw commented on docs/whats-new/2019-09.md:
```diff
@@ -0,0 +1,160 @@
+
+## Architecture guides
+
+- ***new!*** [App startup](../architecture/blazor-for-web-forms-developers/app-startup.md) - Update master with blazor-ebook
```
Instead of listing individual pages, should we simply list the entire e-book?

mairaw commented on docs/whats-new/2019-09.md:
```diff
@@ -0,0 +1,160 @@
+
+## .NET Core desktop
+
+- ***new!*** [Data binding overview in WPF](../desktop-wpf/data/data-binding-overview.md) - Desktop Guide - WPF Initial Release
```
These are actually updates that got moved to the new Desktop guide. Should we remove the new labels?

mairaw commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,60 @@
+    linkLists:
+      - linkListType: overview
+        links:
+          - text: ".NET Docs repositories."
```
```suggestion
          - text: ".NET docs repository"
```

mairaw commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,60 @@
+
+metadata:
+  title: .NET documentation what's new?
+  description: "Welcome to what's new in .NET docs."
```
```suggestion
  description: "Learn about new and updated content in .NET docs."
```

mairaw commented on docs/whats-new/index.yml:
```diff
@@ -0,0 +1,60 @@
+            url: https://github.com/dotnet/core/blob/master/release-notes/README.md
+          - text: ASP.NET Core release notes
+            url: https://docs.microsoft.com/aspnet/core/release-notes/aspnetcore-3.1
+          - text: C# compiler (roslyn) release notes
```
```suggestion
          - text: C# compiler (Roslyn) release notes
```

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,161 @@
+- ***new!*** [Create a template for a control](../desktop-wpf/themes/how-to-create-apply-template.md) - Add desktop wpf article for templates
+- [Styles and templates in WPF](../desktop-wpf/fundamentals/styles-templates-overview.md) - Add desktop wpf article for templates
+
+## .NET framework
```
```suggestion
## .NET Framework
```

mairaw commented on docs/whats-new/2019-09.md:
```diff
@@ -0,0 +1,160 @@
+- ***new!*** [Styles and templates in WPF](../desktop-wpf/fundamentals/styles-templates-overview.md) - Desktop Guide - WPF Initial Release
+- ***new!*** [Overview of XAML resources](../desktop-wpf/fundamentals/xaml-resources-define.md) - Desktop Guide - WPF Initial Release
+
+## .NET framework
```
```suggestion
## .NET Framework
```

mairaw commented on docs/whats-new/2019-10.md:
```diff
@@ -0,0 +1,230 @@
+
+- [Migrating WPF apps to .NET Core](../desktop-wpf/migration/convert-project-from-net-framework.md) - Fix broken links + remove alias links + remove we
+
+## .NET framework
```
```suggestion
## .NET Framework
```

mairaw commented on docs/whats-new/2019-09.md:
```diff
@@ -0,0 +1,160 @@
+- [NextTurn](https://github.com/NextTurn)  - Next Turn
+- [nemrism](https://github.com/nemrism)  - Aymeric A
+- [pkulikov](https://github.com/pkulikov)  - Petr Kulikov
+- [rpetrusha](https://github.com/rpetrusha)  - Ron Petrusha
```
```suggestion
```

mairaw commented on docs/whats-new/2019-09.md:
```diff
@@ -0,0 +1,160 @@
+- [rpetrusha](https://github.com/rpetrusha)  - Ron Petrusha
+- [nschonni](https://github.com/nschonni)  - Nick Schonning
+- [Potapy4](https://github.com/Potapy4)  - Nikita Potapenko
+- [mvelosop](https://github.com/mvelosop)  - Miguel Veloso
```
```suggestion
```

mairaw commented on docs/whats-new/2019-10.md:
```diff
@@ -0,0 +1,230 @@
+- [Youssef1313](https://github.com/Youssef1313)  - Youssef Victor
+- [NextTurn](https://github.com/NextTurn)  - Next Turn
+- [pkulikov](https://github.com/pkulikov)  - Petr Kulikov
+- [rpetrusha](https://github.com/rpetrusha)  - Ron Petrusha
```
he was still Microsoft 

```suggestion
```

mairaw commented on docs/whats-new/2019-10.md:
```diff
@@ -0,0 +1,230 @@
+- [rpetrusha](https://github.com/rpetrusha)  - Ron Petrusha
+- [andresnava](https://github.com/andresnava)  - Andrés Nava
+- [nschonni](https://github.com/nschonni)  - Nick Schonning
+- [mvelosop](https://github.com/mvelosop)  - Miguel Veloso
```
vendor 

```suggestion
```

mairaw commented on docs/whats-new/2019-10.md:
```diff
@@ -0,0 +1,230 @@
+- [JJS](https://github.com/JJS) 
+- [reisenberger](https://github.com/reisenberger)  - Dylan Reisenberger
+- [alavifazel](https://github.com/alavifazel)  - Iman Alavi Fazel
+- [ardalis](https://github.com/ardalis)  - Steve Smith
```
vendor 

```suggestion
```

mairaw commented on docs/whats-new/2019-10.md:
```diff
@@ -0,0 +1,230 @@
+- [Reisclef](https://github.com/Reisclef)  - Richard
+- [rena0157](https://github.com/rena0157)  - Adam Renaud
+- [ReQ6W7rFnaBxyfb](https://github.com/ReQ6W7rFnaBxyfb)  - BR
+- [RobinRH](https://github.com/RobinRH)  - Robin Reynolds-Haertle
```
Microsoft 

```suggestion
```

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,161 @@
+- [nschonni](https://github.com/nschonni)  - Nick Schonning
+- [SeanKilleen](https://github.com/SeanKilleen)  - Sean Killeen
+- [ThePiranha](https://github.com/ThePiranha)  - Sabah Shariq
+- [mvelosop](https://github.com/mvelosop)  - Miguel Veloso
```
```suggestion
```

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,161 @@
+- [Pablosky12](https://github.com/Pablosky12)  - Pablo Marcano
+- [rpetrusha](https://github.com/rpetrusha)  - Ron Petrusha
+- [samtrion](https://github.com/samtrion)  - Martin Stühmer
+- [steveada](https://github.com/steveada)  - Steve Adams
```
vendor

```suggestion
```

mairaw commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,161 @@
+- [kirtisagar](https://github.com/kirtisagar)  - Kirtisagar
+- [lbragaglia](https://github.com/lbragaglia)  - Lorenzo Bragaglia
+- [marekr](https://github.com/marekr)  - Marek Roszko
+- [markrendle](https://github.com/markrendle)  - Mark Rendle
```
vendor 

```suggestion
```

mairaw reviewed (approved):
LGTM but left a few comments for you to consider

tdykstra commented on docs/whats-new/2019-09.md:
```diff
@@ -0,0 +1,160 @@
+- [ML.NET Documentation - Tutorials, API Reference](../machine-learning/index.yml) - Update ML.NET landing page to new design
+- [Tutorial: Generate an ML.NET image classification model from a pre-trained TensorFlow model](../machine-learning/tutorials/image-classification.md) - Remove inaccuracies and improve experience with ML.NET transfer learning tutorial
+
+## .NET Standard
```
Can we put .NET Standard with the two other .NET sections?

tdykstra reviewed (approved):
LGTM, one suggestion.

BillWagner committed:
Apply suggestions from code review

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>

BillWagner commented on docs/whats-new/2019-11.md:
```diff
@@ -0,0 +1,161 @@
+## Apache Spark for .NET
+
+- ***new!*** [Submit a .NET for Apache Spark job to Azure HDInsight](../spark/how-to-guides/hdinsight-deploy-methods.md) - Spark: hdi deployment methods with livy
+- [Customer intent: As a developer, I want to write a simple custom application using .NET for Apache Spark.](../spark/tutorials/get-started.md) - .NET Spark - Getting Started Doc
```
good catch. That's an error in my parsing code for titles. See https://github.com/dotnet/docs/issues/16189#issuecomment-566833114

BillWagner reviewed (commented):
Thanks for the reviews.

All addressed, and doing the build.

BillWagner committed:
respond to feedback.

BillWagner committed:
Use headers instead of highlight

The ***new!*** addition is a bit much.

BillWagner committed:
markdown lint

BillWagner committed:
dropped one title.

BillWagner committed:
final proofread

mairaw commented:
I think no algorithm will be perfect. But as we were reviewing Max’s reddit posts, I think it’s easy for us to identify some oddities right after we finish a month.
